libs/constrained-generators/src/Constrained/Instances.hs

```diff
@@ -74,7 +79,7 @@ instance BaseUniverse fn => Functions (BoolFn fn) fn where
   propagateSpecFun Or (HOLE :? Value (s :: Bool) :> Nil) spec = caseBoolSpec spec (okOr s)
   propagateSpecFun Or (Value (s :: Bool) :! NilCtx HOLE) spec = caseBoolSpec spec (okOr s)
 
-  mapTypeSpec Not _ = typeSpec ()
+  mapTypeSpec Not (SumSpec a b) = typeSpec $ SumSpec b a
```

Comment:
Have we changed the TypeSpec of Bool from () to SumSpec () ()
I didn't notice.
Or is there something more subtle here?

Reply:
Yes, it's now a Generic instance. This change in turn allows us to implement ifElse